src/screens/ProfileScreen/index.js

```diff
@@ -120,7 +120,7 @@ const Header = (props) => {
           <ProfilePicture
             onImageContainerClick={changeImage}
             profilePicPath={dataMain.profile_pic_path}
-            karmaScore={dataMain.karma_score}
+            karmaScore={dataMain.is_karma_unlocked ? dataMain.karma_score : 0}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change made here is a good one, as it ensures that the `karmaScore` prop only receives `dataMain.karma_score` if `dataMain.is_karma_unlocked` is true. Otherwise, it sets `karmaScore` to 0. This prevents potential issues where `karma_score` might be undefined or not applicable when `is_karma_unlocked` is false. However, it would be better to ensure that `dataMain.karma_score` is defined before using it. If it's undefined, this could potentially cause errors.

```diff
- karmaScore={dataMain.is_karma_unlocked ? dataMain.karma_score : 0}
+ karmaScore={dataMain.is_karma_unlocked && dataMain.karma_score ? dataMain.karma_score : 0}
```
This way, we are checking both `is_karma_unlocked` and `karma_score` before assigning the value.


<!-- This is an auto-generated comment by OSS CodeRabbit -->